operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependentResource.java

```diff
@@ -152,51 +129,72 @@ public R update(R actual, R target, P primary, Context<P> context) {
           actual.getMetadata().getResourceVersion());
     }
     R updatedResource;
+    addMetadata(false, actual, target, primary);
     if (useSSA(context)) {
       updatedResource = prepare(target, primary, "Updating")
           .fieldManager(context.getControllerConfiguration().fieldManager())
           .forceConflicts().serverSideApply();
     } else {
       var updatedActual = updaterMatcher.updateResource(actual, target, context);
-      updatedResource = prepare(updatedActual, primary, "Updating").update();
+      updatedResource = prepare(updatedActual, primary, "Updating").replace();
```

Comment:
Actually I think it was just confusing for the rebase because my changes included toggling this from replace to update, then back to replace.  Once you rebased next your change from replace to update logically came before my changing it back to replace. 

I'll start squashing my changes before rebasing to prevent that sort of thing.